server/src/main/java/org/candlepin/controller/SuspendModeTransitioner.java

```diff
@@ -0,0 +1,232 @@
+    private static Logger log = LoggerFactory.getLogger(SuspendModeTransitioner.class);
+    /**
+     * Factor by which the delay is going to be increased with each failed attempt to
+     * reconnect to the Qpid Broker. For example if set to 3 and initialDelay set to
+     * 5. The following will be the delay interval between the executions of this
+     * transitioner:
+     *
+     * When failedAttempts = 0, then delay will be 5
+     * When failedAttempts = 1, then the delay will be 15
+     * When failedAttempts = 2, then the delay will be 45
+     * etc.
+     *
+     */
+    private int delayGrowth;
+    /**
+     * Delay in seconds. SuspendModeTransitioner will wait this amount
+     * of seconds between periodic checks for Qpid connectivity.
+     */
+    private int initialDelay;
+    /**
+     * Maximum delay that this transitioner can wait between executions
+     */
+    private int maxDelay = 0;
+    private int failedAttempts = 0;
+    /**
+     * Single threaded periodic task.
+     */
+    private ScheduledExecutorService execService;
+    private ModeManager modeManager;
+    private QpidQmf qmf;
+    private QpidConnection qpidConnection;
+
+    @Inject
+    public SuspendModeTransitioner(Configuration config, ScheduledExecutorService execService) {
+        this.execService = execService;
+        delayGrowth = config.getInt(ConfigProperties.QPID_MODE_TANSITIONER_DELAY_GROWTH);
+        initialDelay = config.getInt(ConfigProperties.QPID_MODE_TRANSITIONER_INITIAL_DELAY);
+        maxDelay = config.getInt(ConfigProperties.QPID_MODE_TRANSITIONER_MAX_DELAY);
+    }
+
+    /**
+     * Other dependencies are injected using method injection so
+     * that Guice can handle circular dependency between SuspendModeTransitioner
+     * and the QpidConnection
+     */
+    @Inject
+    public void setModeManager(ModeManager modeManager) {
+        this.modeManager = modeManager;
+    }
+
+    @Inject
+    public void setQmf(QpidQmf qmf) {
+        this.qmf = qmf;
+    }
+
+    @Inject
+    public void setQpidConnection(QpidConnection qpidConnection) {
+        this.qpidConnection = qpidConnection;
+    }
+
+    /**
+     * Enables to run the transitioning logic periodically.
+     */
+    public void startPeriodicExecutions() {
+        log.info("Starting Periodic Suspend Mode Transitioner " +
+            "with delay grow factor of {}s and initial delay {}s ",
+            delayGrowth, initialDelay);
+        schedule();
+    }
+
+    /**
+     * Schedules next execution of the Suspend Mode check. The delay, in seconds,
+     * grows as the failedAttempts grow. The formula is:
+     *
+     *  delay = initialDelay * (delayGrowth ^ failedAttempts)
+     *
+     *  Maximal delay is configurable, -1 means there is no bound
+     *
+     *  When delayGrowth is set to 1, the delay wont grow.
+     */
+    private void schedule() {
+        int delay = initialDelay * (int) Math.pow(delayGrowth, failedAttempts);
```

Comment:
Changed the logic so that the backoff is linear now. 

delay = initialDelay + (delayGrowth * failedAttempts)

Where default values are:
initalDelay = 10 seconds
delayGrowth = 10 seconds

maxDelay changed to 5 minutes